File: typsys_scp_closures.js
// showing type system
function out_fn() {

    let x = 22; // Number
    let y = [1, 2, 3]; // Array
    let z = true; // Boolean
    let out_var = "outer"
    
    console.log("x is of type:", typeof x); 
    console.log("y is of type:", typeof y); 
    console.log("z is of type:", typeof z); 
    console.log("out_var is of type:", typeof out_var); 
    
    return function in_fn() {
        // Inner function has access to the outer scope variable
        return 'Accessing outer variable: ' + out_var

    };
}

const closure = out_fn();
console.log(closure());  // Calling the closure function

File: typsys_scp_closures.py
def out_fn():
    x = 20 # Integer
    y = [1, 2, 3]  # List
    z = 1.4  # Float
    out_var = "Outer" #string
    print("x is of type:", type(x))
    print("y is of type:", type(y))
    print("z is of type:", type(z))
    print("out_var is of type:", type(out_var))

    def in_fn():
        # Accessing outer variable from the outer scope
        return "Inner function can access:",out_var
    
    return in_fn  # Returning the closure

closure = out_fn()
print(closure())  # Calling the closure function